### ISSUE TITLE
add support for UniProt variants annotations

### ISSUE THREAD

**Issue by fkaiserbio** [singa-bio/singa#83]
- **required adaptions**: implement XML parsing in `UniProtContentHandler`, add new feature class
- we should add the support for genetic variants mapping in the UniProt database
- extraction of dbSNP identifier from `description` attribute
- for this the parsing of an additional feature is necessary:
```
<feature type="sequence variant" description="In AMYL-TTR; amyloid polyneuropathy; dbSNP:rs121918091." id="VAR_007575" evidence="25 57 70">
  <original>F</original>
  <variation>L</variation>
  <location>
    <position position="84"/>
  </location>
</feature>
```
- along with this, parsing of literature evidence should be implemented, specified by the `evidence` attribute of the "sequence variation" feature:
```
<reference key="1">
  <citation type="journal article" date="1984" name="Biochem. Biophys. Res. Commun." volume="124" first="558" last="564">
  <title>
    Cloning and sequence analysis of cDNA for human prealbumin.
  </title>
  <authorList>
...
</reference>
```
>  f.e. `evidence="1 2"` would refer to references 1 and 2. For now, the parsing of the doi should be sufficient.
- for a good example see [P02766](https://www.uniprot.org/uniprot/P02766.xml)

**Comment by cleberecht** [comment]
implemented support for sequence variants in commit 4fd7d42 (branch `feature/mutation-variation-parsing`)

- added new feature class `SequenceVariants`
- the following features of sequence variations are extracted: id, description, evidence, original, variation, and location
- dbSNP is not always annotated in description, neither is description in general. If you explicitly require it I would recommend implementing an `Identifier` (e.g. see `ECNumber`), registering it in the `IdentifierPatternRegistry` and adding the corresponding attribute to the `SequenceVariant` class
- evidences are parsed, previous `FeatureOrigins` are now reimplemented as `Evidence`, containing the evidence type, name (AuthorYear if literature), and preferably DOI, if no DOI is available pubmed or nothing otherwise

You can get the variations using:
``` java
Protein transthyretin = UniProtParserService.parse("P02766");
List<SequenceVariant> sequenceVariants = transthyretin.getFeature(SequenceVariants.class).getFeatureContent();
```
Please report errors and bugs in this thread.

**Comment by fkaiserbio** [comment]
- seems to work partly, test `SingleFileChannelMembraneTransportTest` fails
- it seems that there are some index access issues when parsing the `AminoAcidFamily` from the sequence by its one-letter code

**Comment by cleberecht** [comment]
should be fixed in bf99b60

**Comment by fkaiserbio** [comment]
merged in development https://github.com/singa-bio/singa/commit/093fca42cb31ffac09b01ca911a737b5ae1dca8c
